### ISSUE TITLE
Image version reported by StackRox does not always what is deployed, after the image has updated

### ISSUE THREAD

**Issue by danekantner** [stackrox/stackrox#6737]
Image version reported by StackRox does not always what is deployed, after the image has updated

StackRox has generated a finding and shows the image tag as `us.gcr.io/tcr/cvd-trn:1ad0d4ed267ac382ff1b440570c7b9ca059efeac` however the actual image running is correctly shown in GKE as the newer tagged image `us.gcr.io/tcr/cvd-trn:3dd962e809561ecb73705497b6ba627c106fb467` -- this image version was the result of a code rebuild but did not actually address the findings. The expecation is that StackRox should update the original alert i.d. with the latest tag 3dd, or close out the original alert i.d. and create a new alert for that deployment with the new image i.d., but it has done neither in this case and just remains stagnant showing the old version as if it is still deployed.

New images are released and the deployment is updated via `kubectl` set image commands, generally following the pattern:

```
kubectl rollout pause "deployment/$DEPLOYMENT" --namespace "$NAMESPACE"

 # The deadline is the K8s value for the deadline a deployment must succeed, this value is our main timeout value
deadline="$(kubectl get deployment $DEPLOYMENT --namespace $NAMESPACE -o jsonpath='{.spec.progressDeadlineSeconds}')"

kubectl set image "deployment/$DEPLOYMENT" "$CONTAINER_NAME=$IMAGE:$IMAGE_TAG" --namespace "$NAMESPACE" 
kubectl set image "deployment/$DEPLOYMENT" "$INIT_CONTAINER_NAME=$IMAGE:$IMAGE_TAG" --namespace "$NAMESPACE"

 log "$(bold "Waiting up to $deadline seconds for rollout to succeed, until $(date -d "+$deadline seconds"). This value is your deployment's progressDeadlineSeconds value (Progress deadline value in the GCP K8s console's details page for your deployment).")"
  kubectl rollout resume "deployment/$DEPLOYMENT" --namespace "$NAMESPACE" >&2
```

(this is an automated process within our CI/CD system). 

This process has not changed on our end in several years, and we have had our pipelines set up and StackRox and issues have been consistently updating as expected until now.

StackRox API and UX both still show the old image tag, despite the new deployment successfully have been updated in GKE and GKE shows the new image tag everywhere for both the main container and init container versions.

```
"containers": [
            {
                "image": {
                    "id": "sha256:f66b8d9ef6f51008b92304ee0e5cfcd1c29cc5ee513b973be3cf76a9ff498b9e",
                    "name": {
                        "registry": "us.gcr.io",
                        "remote": "tcr/workflows-cvd-trn",
                        "tag": "1ad0d4ed267ac382ff1b440570c7b9ca059efeac",
                        "fullName": "us.gcr.io/tcr/cvd-trn:1ad0d4ed267ac382ff1b440570c7b9ca059efeac"
                    },
                    "notPullable": false,
                    "isClusterLocal": false
                },
                "name": "main"
            }
        ],
```

and it is still listed as ACTIVE state:
```
    "processViolation": null,
    "enforcement": null,
    "time": "2023-06-17T07:47:01.076485872Z",
    "firstOccurred": "2023-03-06T20:21:55.016633955Z",
    "resolvedAt": null,
    "state": "ACTIVE",
    "snoozeTill": null
```

The image version reported in the UX matches what the API returns.

This example with this issue was originally generated via StackRox 3.74.0, and it was subsequently upgraded to 4.0.0, and the kubectl set image commands were ran a few weeks after that under 4.0.0.

**Comment by danekantner** [comment]
This issue resolved itself by restarting the sensors on all of the clusters

**Comment by kurlov** [comment]
Thank you for reporting the issue.
I am closing it since it's resolved but I guess it still might be interesting for @stackrox/maple team.
Please, feel free to reopen the issue if it happens again

**Comment by vikin91** [comment]
@Maddosaurus shall we have a ticket for this issue or has it been resolved recently?

**Comment by danekantner** [comment]
This is still not resolved AFAIK but I thought there was an actual issue open for it already

**Comment by Maddosaurus** [comment]
I've checked our internal tracking and wasn't able to find a ticket for this.
I've created ROX-20810 and assigned it to the responsible eng team. 
Apologies that this got lost, @danekantner!

**Comment by danekantner** [comment]
Thank you, for reasons even I don't understand I had apparently created a duplicate of this issue (https://github.com/stackrox/stackrox/issues/6995) and in that issue was a reference to  https://github.com/stackrox/stackrox/pull/7304 that had been opened and is awaiting merging. IDK if that will fix it or not.

**Comment by connorgorman** [comment]
@danekantner I believe the fix for this has been merged in for the 4.4 release. Will close this ticket. Please reopen if you think otherwise
